scanpipe/pipes/d2d.py

```diff
@@ -114,6 +125,55 @@ def get_best_path_matches(to_resource, matches):
     return matches
 
 
+def load_ecosystem_config(pipeline, options):
+    """
+    Add ecosystem specific configurations for each ecosystem selected
+    as `options` to the `pipeline`.
+    """
+    configs_by_ecosystem = {
+        ecosystem.ecosystem_option: ecosystem for ecosystem in ECOSYSTEM_CONFIGS
+    }
+
+    # Add default configurations which are common accross ecosystems
+    add_ecosystem_config(
+        pipeline=pipeline,
+        configs_by_ecosystem=configs_by_ecosystem,
+        selected_option="Default",
+    )
+
+    # Add configurations for each selected ecosystem
+    for selected_option in options:
+        if selected_option not in configs_by_ecosystem:
+            continue
+
+        add_ecosystem_config(
+            pipeline=pipeline,
+            configs_by_ecosystem=configs_by_ecosystem,
+            selected_option=selected_option,
+        )
+
+
+def add_ecosystem_config(pipeline, configs_by_ecosystem, selected_option):
+    d2d_pipeline_configs = [
+        "purldb_package_extensions",
+        "purldb_resource_extensions",
+        "deployed_resource_path_exclusions",
+    ]
+
+    ecosystem_config = configs_by_ecosystem.get(selected_option)
+
+    for pipeline_config in d2d_pipeline_configs:
+        config_value = getattr(ecosystem_config, pipeline_config)
+        pipeline_config_value = getattr(pipeline, pipeline_config)
+        if config_value:
+            if not pipeline_config_value:
+                new_config_value = config_value
+            else:
+                new_config_value = pipeline_config_value.extend(config_value)
+
+            setattr(pipeline, pipeline_config, new_config_value)
```

Comment:
Not ideal to set values from all the way down here, shouldn't we return those to a higher location that will explicitly set the values?

Reply:
I'm not sure if I can change this, as this is the cleanest way where we can update a bunch of configuration attributes based on all the ecosystems/options selected. If we return these values, there's no use of this function at all, which refactors and gets rid of duplicate code.